File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        n=len(heights)
        maxContent=0
        l=0
        r=n-1
        while l<r:
            w=r-l
            h=min(heights[l],heights[r])
            content=w*h
            maxContent=max(content,maxContent)
            if heights[l]<heights[r]:
                l+=1
            else:
                r-=1
        
        return maxContent


        

File: Data Structures & Algorithms/min-cost-climbing-stairs/submission-0.cpp
class Solution {
public:
int func(int i,vector<int> &cost,int c){
    if(i>cost.size()){
          return c;
    }
    c+=cost[i];
    int lc=func(i+1,cost,c);
    
    int rc=func(i+2,cost,c);
     return min(lc,rc);

}
    int minCostClimbingStairs(vector<int>& cost) {
        return min(func(0,cost,0),func(1,cost,0));
        
    }
};
